lib/api/src/externals/memory.rs

```diff
@@ -84,7 +85,7 @@ impl Memory {
     }
 
     /// Returns the size (in bytes) of the `Memory`.
-    pub fn data_size(&self) -> usize {
+    pub fn data_size(&self) -> u32 {
```

Comment:
This change seems fine on `Memory` a concrete type, but for something like a Memory trait, we probably want `u64`.  Wasm64 will eventually ship with more than 4gb of memory

Reply:
Done.